agent/xds/routes.go

```diff
@@ -173,6 +262,8 @@ func makeUpstreamRouteForDiscoveryChain(
 		return nil, fmt.Errorf("missing first node in compiled discovery chain for: %s", chain.ServiceName)
 	}
 
+	var lb *structs.EnvoyLBConfig
```

Comment:
This variable seems like it is better suited for living entirely within the case blocks, since it's not used in long-distance fashion.